npm/qsharp/ux/circuit.tsx

```diff
@@ -12,56 +13,157 @@ import { CircuitProps } from "./data.js";
 const MAX_OPERATIONS = 10000;
 const MAX_QUBITS = 1000;
 
-/* This component is shared by the Python widget and the VS Code panel */
+// This component is shared by the Python widget and the VS Code panel
 export function Circuit(props: { circuit: qviz.Circuit }) {
+  const circuit = props.circuit;
+  const unrenderable =
+    circuit.qubits.length === 0 ||
+    circuit.operations.length > MAX_OPERATIONS ||
+    circuit.qubits.length > MAX_QUBITS;
+
+  return (
+    <div>
+      {unrenderable ? (
+        <Unrenderable
+          qubits={props.circuit.qubits.length}
+          operations={props.circuit.operations.length}
+        />
+      ) : (
+        <ZoomableCircuit circuit={props.circuit} />
+      )}
+    </div>
+  );
+}
+
+function ZoomableCircuit(props: { circuit: qviz.Circuit }) {
   const circuitDiv = useRef<HTMLDivElement>(null);
+  const [zoomLevel, setZoomLevel] = useState<number | undefined>(undefined);
```

Comment:
Shouldn't this just be initialized to `100` and not have `undefined` in the type?

Reply:
`undefined` controls the visibility of the zoom control, so we don't show it before the circuit has finished rendering. But yeah, I think I can still simplify this by using the existing `rendering` flag and removing `undefined`.